graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -107,26 +130,149 @@ def rename_schema(
     query_type = get_query_type_name(schema)
     scalars = get_scalar_names(schema)
 
-    # Rename types, interfaces, enums
-    schema_ast, reverse_name_map = _rename_types(schema_ast, renamings, query_type, scalars)
+    _validate_renamings(schema_ast, renamings, query_type)
+
+    # Rename types, interfaces, enums, unions and suppress types, unions
+    schema_ast, reverse_name_map = _rename_and_suppress_types(
+        schema_ast, renamings, query_type, scalars
+    )
     reverse_name_map_changed_names_only = {
         renamed_name: original_name
         for renamed_name, original_name in six.iteritems(reverse_name_map)
         if renamed_name != original_name
     }
 
-    # Rename query type fields
-    schema_ast = _rename_query_type_fields(schema_ast, renamings, query_type)
+    # Rename and suppress query type fields
+    schema_ast = _rename_and_suppress_query_type_fields(schema_ast, renamings, query_type)
     return RenamedSchemaDescriptor(
         schema_ast=schema_ast,
         schema=build_ast_schema(schema_ast),
         reverse_name_map=reverse_name_map_changed_names_only,
     )
 
 
-def _rename_types(
+def _validate_renamings(
+    schema_ast: DocumentNode, renamings: Mapping[str, Optional[str]], query_type: str
+) -> None:
+    """Validate the renamings argument before attempting to rename the schema.
+
+    Check for fields with suppressed types or unions whose members were all suppressed. Also,
+    confirm renamings contains no enums, interfaces, or interface implementation suppressions
+    because that hasn't been implemented yet.
+
+    The input AST will not be modified.
+
+    Args:
+        schema_ast: schema that we're returning a modified version of
+        renamings: maps original type name to renamed name or None (for type suppression). Any name
+                   not in the dict will be unchanged
+        query_type: name of the query type, e.g. 'RootSchemaQuery'
```

Comment:
(for future PR) we should be able to load this from the AST, since the schema starts with
```
schema {
    query: RootSchemaQuery
    ...
}

...
```